tools/run_tests/performance/loadtest_template.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+# Copyright 2021 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script generates a load test configuration template from a collection of
+# load test configurations.
+#
+# The following example generates a basic template from the example configs
+# in https://github.com/grpc/test-infra/tree/master/config/samples:
+#
+# $ ./tools/run_tests/performance/loadtest_template.py \
+#     -i ../test-infra/config/samples/*.yaml \
+#     -o ./tools/run_tests/performance/templates/basic_template.yaml \
+#     --name basic_template
+
+import argparse
+import sys
+
+from typing import Any, Dict, Iterable, Mapping, Type
+
+import yaml
+
+import loadtest_config
+
+TEMPLATE_FILE_HEADER_COMMENT = """
+# Template generated from load test configurations by loadtest_template.py.
+#
+# Configuration templates contain client and server configurations for multiple
+# languages, and may contain template substitution keys. These templates are
+# used to generate load test configurations by selecting clients and servers for
+# the required languages. Load test configuration generation is performed by
+# loadtest_config.py. See documentation below:
+# https://github.com/grpc/grpc/blob/master/tools/run_tests/performance/README.md
+"""
+
+
+def loadtest_template(
+        input_file_names: Iterable[str],
+        metadata: Mapping[str, Any],  # Hello
+        inject_client_pool: bool,
+        inject_server_pool: bool,
+        inject_big_query_table: bool,
+        inject_timeout_seconds: bool,
+        inject_ttl_seconds: bool) -> Dict[str, Any]:
```

Comment:
If I understand this right, from line 49 to line 55 describes injectable fields of the generated template. After the templated generated, these are the only fields can be updated through loadtest_config.py.  For prebuilt images, we also need a filed `image` under `run` container so that we can generate load test configurations using pre-built images.

Reply:
I would expect this to be added to the input files to loadtest_template.py. If the inputs already have substitution keys, these will be passed unchanged to the output. The "--inject_"... options are only for a subset of options that are likely to be useful for all kinds of tests – running on different pools, or running as part of a larger set of tests that may require a larger timeout. Image locations are at the same time too specific to one case (for instance, prebuilt images) and unlikely to change when you run the test multiple times or under different conditions.